lib/chef/resource/windows_share.rb

```diff
@@ -306,13 +305,13 @@ def permissions_need_update?(type)
           property_name = "#{type}_users"
 
           # brand new share, but nothing to set
-          return false if current_resource.nil? && new_resource.send(property_name).empty?
+          return false if current_resource.nil? && instance_variable_get("@#{property_name}").empty?
 
           # brand new share with new permissions to set
-          return true if current_resource.nil? && !new_resource.send(property_name).empty?
+          return true if current_resource.nil? && !instance_variable_get("@#{property_name}").empty?
 
           # there's a difference between the current and desired state
-          return true unless (new_resource.send(property_name) - current_resource.send(property_name)).empty?
+          return true unless (instance_variable_get("@#{property_name}") - current_resource.send(property_name)).empty?
```

Comment:
We've moved away from using the instance vars, and instead use the methods on the right object. This feels like a regression.

Reply:
I was backporting this fix to chef16 and def `new_resource_users` is using those instance variables instead of setting the properties on new_resource due to which `new_resource.send(property_name)` never actually has the correct values. So it then results in test failure https://buildkite.com/chef-oss/chef-chef-chef-16-verify/builds/1448#0185bfd7-35c0-4015-bda2-0363cfff6bf5/302-1202. It's working here because of `instance_variable_get`

Ref https://github.com/chef/chef/pull/13496/files#diff-4a31658e49ff21d95272bb16f9375a8a929d96388ba81eb837f7989c78b420daR317-R323